src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts

```diff
@@ -0,0 +1,45 @@
+import { D2Api } from "@eyeseetea/d2-api/2.36";
+import log from "utils/log";
+import { PermissionFixerUserGroupExtended } from "domain/entities/user-monitoring/permission-fixer/PermissionFixerUserGroupExtended";
+import { PermissionFixerUserGroupRepository } from "domain/repositories/user-monitoring/permission-fixer/PermissionFixerUserGroupRepository";
+import { Async } from "domain/entities/Async";
+
+export class PermissionFixerUserGroupD2Repository implements PermissionFixerUserGroupRepository {
+    constructor(private api: D2Api) {}
+    async getByIds(groupsIds: string[]): Async<PermissionFixerUserGroupExtended> {
+        log.info(`Get metadata: All groups`);
+
+        //todo use d2api filters
+        const responses = await this.api
+            .get<UserGroups>(
+                `/userGroups?filter=id:in:[${groupsIds.join(",")}]&fields=id,name,users,*&paging=false.json`
+            )
+            .getData();
+
+        const userGroups = responses["userGroups"][0] ?? undefined;
+        if (userGroups) {
+            return userGroups;
+        } else {
+            throw new UserGroupNotFoundException("Error getting user groups: " + groupsIds.join(","));
+        }
+    }
```

Comment:
this should be in `RunUserPermissionUseCase`. `getByIds` usually gets multiple userGroups, hence the array.
As said in the other comment, i think this should become a get with a single id. Because its only used in that way.